validator/client/validator.go

```diff
@@ -86,6 +86,7 @@ type validator struct {
 	graffiti                           []byte
 	voteStats                          voteStats
 	graffitiStruct                     *graffiti.Graffiti
+	graffitiOrderedIndex               uint
```

Comment:
Are we using this?

Reply:
Yes, a place to store what index we are up to: https://github.com/prysmaticlabs/prysm/pull/8189/files#diff-20de6d51bd5bdb24dcd8f3634f1a06f933e6b5049c9502bb0c634b5ec6304724R306

Reply:
Well, it's not defined in the Graffiti struct, so how is it used?

Reply:
oh, you mean loading the `starting_index` from the YAML? The original proposal is not having that and just the `ordered` which contains a list of strings to use. I tend to keep it similar to `random` (not nested in `ordered.list`) and doubt if people will really go to the YAML and change the `starting_index`, and in case of auto restarting from crash (infrequent though) it will also just start from last set `starting_index`.

So I just assume it is fine for `graffitiOrderedIndex` always start from `0` whenever `validator` restarts (users can also reorder the items if they really want to), keeping it less config.

What do you think?

Reply:
We originally had the same `ordered` implementation as well with the same intend but decided to scrap it. Im personally think it's bad to always start from `0` or expect validators to manually adjust index. Proposal is a very rare and exciting event, the node operators will get disappointed finding out graffiti is not the one intended to be

Reply:
@terencechain Agree, it is bad to be unpredictable for this. Do you think we can persist the last index to the validator db? The db seems is for chain info so far so is it appropriate to store other kinds of data like this?

Reply:
Yes. I think that's the best path forward if we do pursue an `ordered` option
From the look of it, it seems like there's enough interest to push it forward

Reply:
I don't think a DB option is a great way to do it. How does the user know what the last index was or how to reset it?

I thought we had decided on setting the index in the yaml and the user could update it from time to time as needed.

Reply:
Since this comment was linked to I just want to echo my suggestion below... `ordered` is useful to me in order to emit graffiti messages that aren't repeated. At the moment I use `random` and every so often remove emitted entries, which I guess would be a similar process to ordered. I don't see any value of having the `index` in the YML file compared to just commenting out the first X entries though.

One easy solution to your "last index reset" problem is to store the index and a hash of the file in the database. That way it'd be trivial to reset the index whenever the file is updated (as a user would expect) and not reset it whenever the server is restarted (which should be a no-op from a behavior standpoint.)

Reply:
To be clear I'm not the original author of the change, but would like to get involved in the project so happy to try my hand at implementing that feature if @pinglamb would like me to.

Reply:
@qix Feel free to work on that!

Reply:
@pinglamb just pushed https://github.com/prysmaticlabs/prysm/pull/8482 up, hopefully it'll get a :ok: